File: P2/QytetetRuby/lib/casilla.rb
#ENCODING: UTF-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module QytetetRuby
class Casilla
  private_class_method :new
  attr_reader :numeroCasilla, :coste, :titulo, :tipo
  attr_writer :titulo
  private :titulo, :titulo=
  def initialize(numeroCasilla_,coste_,tipo_,titulo_)
    @numeroCasilla = numeroCasilla_
    @coste = coste_
    @tipo = tipo_
    @titulo = titulo_
  end
  def self.noTipoCalle(numeroCasilla_,coste_,tipo_)
    new(numeroCasilla_,coste_,tipo_,nil)
  end
  def self.tipoCalle(numeroCasilla_,tipo_,titulo_)
    new(numeroCasilla_,titulo_.precioCompra,tipo_,titulo_)
  end
  
  def asignarPropietario(jugador)
    raise NotImplementedError
  end
  def pagarAlquiler
    raise NotImplementedError
  end
  def propietarioEncarcelado
    raise NotImplementedError
  end
  def soyEdificable
    raise NotImplementedError
  end
  def tengoPropietario
    raise NotImplementedError
  end
  
  
  
  
  def to_s
    "Numero Casilla : #{@numeroCasilla}\n
     Coste: #{@coste} \n
     Tipo: #{@tipo}\n
     Titulo Propiedad: #{@titulo.to_s}\n
     "
  end
end
end

File: QytetetJava/src/modeloqytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author marquirj
 */
public class Casilla {
    private int numeroCasilla;
    private int coste;
    private TipoCasilla tipo;
    private TituloPropiedad titulo=null;
    Casilla(int numeroCasilla, int coste, TipoCasilla tipo){
        this.numeroCasilla=numeroCasilla;
        this.coste=coste;
        this.tipo=tipo;
        this.titulo=null;
        
    }
    Casilla(int numeroCasilla, TipoCasilla tipo, TituloPropiedad titulo){
        this.numeroCasilla=numeroCasilla;
        this.coste=titulo.getPrecioCompra();
        this.tipo=tipo.CALLE;
        setTitulo(titulo);
    }

    public int getNumeroCasilla() {
        return numeroCasilla;
    }

    public int getCoste() {
        return coste;
    }

    public TipoCasilla getTipo() {
        return tipo;
    }

    public TituloPropiedad getTitulo() {
        return titulo;
    }
    private void setTitulo(TituloPropiedad titulo){
        this.titulo=titulo;
    }
    @Override
    public String toString(){
        String tit;
        if(this.titulo==null){
            tit="Carcel";
        }else
            tit=this.titulo.toString();
        return "Numero casilla: " + this.numeroCasilla + 
                "\nCoste: " + this.coste + "\nTipo Casilla: " + this.tipo + 
                "\nTitulo de la propiedad: " + tit + "\n\n";
    }
}
